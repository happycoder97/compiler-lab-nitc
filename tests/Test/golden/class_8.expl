class Rec {
    Rec r;
    int init(int n) {
        self.r = null;
        if(n > 0) then
            self.r = new(Rec);
            self.r.init(n-1);
        endif
    }

    int print() {
        if(self.r != null) then
            write("FOO");
            self.r.print();
        endif
    }
}

int main() {
    initialize();
    Rec r;
    r = new(Rec);
    r.init(5);
    r.print();
    write("DONE");
}
