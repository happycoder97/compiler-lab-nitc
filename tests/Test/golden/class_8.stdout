FOO
FOO
FOO
FOO
FOO
DONE
